### PR TITLE
Drop PyTorch 1.8 support

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

With the 1.7 release, support for torch 1.8 is dropped.

### Does your PR introduce any breaking changes? If yes, please list them.

See title.

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

cc @borda @carmocca @akihironitta

### CODE REVIEW

**Review by Borda — DISMISSED** [review]
1.8 is LTS so disagree with this... :/

**Comment by akihironitta** [comment]
@Borda @carmocca Do you know when PyTorch 1.8 will end its life as LTS? If it's planned to be supported for a certain period of time (6 more months? one more year?) after the PL 1.7.0 release, we should maybe keep testing with PyTorch 1.8.

**Comment by carmocca** [comment]
All CI passed but TPU. I forgot to update XLA :/

File: .azure-pipelines/gpu-tests.yml [hunk 1]
```diff
@@ -20,8 +20,6 @@ jobs:
   - job: pytest
     strategy:
       matrix:
-        'PyTorch - LTS':
-          image: "pytorchlightning/pytorch_lightning:base-cuda-py3.7-torch1.8"
```
Comment (Borda) [inline .azure-pipelines/gpu-tests.yml:24]: Let's then test at least 1.9

Comment (carmocca) [reply .azure-pipelines/gpu-tests.yml:24]: One advantage is to reduce the CI impact now. We can do this in a follow-up

**Pull request merged**